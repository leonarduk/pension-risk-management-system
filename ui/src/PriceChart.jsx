import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend,
} from 'chart.js';
import { Line } from 'react-chartjs-2';

ChartJS.register(
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend
);

export default function PriceChart({ labels, data, label }) {
  const chartData = {
    labels,
    datasets: [
      {
        label,
        data,
        borderColor: 'rgb(75, 192, 192)',
        backgroundColor: 'rgba(75, 192, 192, 0.2)',
      },
    ],
  };

  return <Line data={chartData} />;
}
