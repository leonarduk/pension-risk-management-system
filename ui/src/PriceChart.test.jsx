import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { expect, test, vi, afterEach } from 'vitest';
import PriceChart from './PriceChart.jsx';

afterEach(() => {
  cleanup();
});

vi.mock('react-chartjs-2', () => ({
  Line: (props) => <pre data-testid="line-chart">{JSON.stringify(props)}</pre>,
}));

test('renders line chart with provided data', () => {
  const labels = ['Jan', 'Feb'];
  const data = [1, 2];
  render(<PriceChart labels={labels} data={data} label="Test Label" />);
  const chart = screen.getByTestId('line-chart');
  expect(chart.textContent).toContain('"labels":["Jan","Feb"]');
  expect(chart.textContent).toContain('"label":"Test Label"');
  expect(chart).toMatchSnapshot();
});

test('handles empty datasets gracefully', () => {
  render(<PriceChart labels={[]} data={[]} label="Empty" />);
  expect(screen.getByTestId('line-chart')).toBeTruthy();
});
