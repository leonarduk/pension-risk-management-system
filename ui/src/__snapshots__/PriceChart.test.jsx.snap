// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders line chart with provided data 1`] = `
<pre
  data-testid="line-chart"
>
  {"data":{"labels":["Jan","Feb"],"datasets":[{"label":"Test Label","data":[1,2],"borderColor":"rgb(75, 192, 192)","backgroundColor":"rgba(75, 192, 192, 0.2)"}]}}
</pre>
`;
